### DISCUSSION TITLE
Cannot create window when executing simulate binary in linux:22.04 container in m1 macbook pro

### DISCUSSION THREAD

**Discussion by jeeseop** [google-deepmind/mujoco#1999] [Asking for Help]
**How to ask for help**

First, read our quick guide to
[asking good questions](https://github.com/google-deepmind/mujoco#asking-questions).
Below is a template for you to use:

Hi,

I'm a researcher and I'm trying to use MuJoCo for the research.

I'm looking for some help with docker setup in M1 Macbook pro.

Here is a model which explains my question:

Here is the docker-compose.yml file that I am currently using.
============
name: ubuntu
services:
  for_mujoco:
    image: ubuntu:22.04 
    #platform: linux/amd64
    command: /bin/bash
    volumes:
      - /dev:/dev
    environment:
      - DISPLAY=host.docker.internal:0
    privileged: true
    ipc: host
    shm_size: 256m
    network_mode: host
    stdin_open: true
    tty: true
============

I can run xeyes or xclock when using this container with XQuartz in m1 macbook pro.
However, I cannot create a window when executing mujoco's 'simulate' binary in this container in m1 macbook pro.

Here I installed:
* cmake
* build-essential
* libglfw3-dev

This is the full message that I've got when executing 'simulate' binary:

root@docker-desktop:/mujoco-3.1.4/build/bin# ./simulate 
MuJoCo version 3.1.4
ERROR: could not create window

Press Enter to exit ...

**Comment by granstrom** [comment]
I get the same error message when running `simulate` on a remote machine from a MacBook Air M2 with XQuartz where e.g. `xeyes` works fine.

**Comment by dreamot** [comment]
Hello, have you solved this problem? I seem to be having a similar issue, on ubuntu24.04 in Vmare
<img width="568" alt="dd" src="https://github.com/user-attachments/assets/53c70cdd-9ad8-4b23-86f0-a4fa098aca4f" />

**Reply by simbaforrest** [reply]
I just met the same problem and solved it following this:
https://github.com/Farama-Foundation/Gymnasium/issues/755#issuecomment-2825928509
